### PR TITLE
KAA-1351: "Key platform features" documentation page

### PR DESCRIPTION
**Pull request by vzhukovskyi**

### CODE REVIEW

**Comment by kaa-leeroy** [comment]
Can one of the admins verify this patch?

**Comment by rasendubi** [comment]
@kaa-leeroy add to whitelist

File: doc/Programming-guide/Key-platform-features/index.md [hunk 1]
```diff
@@ -5,4 +5,18 @@ permalink: /:path/
 sort_idx: 30
 ---
 
-List platform features and link to subpages. Mention active load balancing and refer to admin guide.
+This guide describes all major features of Kaa and how to effectively use them in IoT applications.
```
Comment (akokhanovskyi) [inline doc/Programming-guide/Key-platform-features/index.md]: remove "all"

File: doc/Programming-guide/Key-platform-features/index.md [hunk 2]
```diff
@@ -5,4 +5,18 @@ permalink: /:path/
...
+| **[Configuration management]({{root_url}}Programming-guide/Key-platform-features/Configuration-management/)** | Allows you to create and distribute configuration data to endpoints.
+| **[Data collection]({{root_url}}Programming-guide/Key-platform-features/Data-collection/)** | Allows you to collect data records (logs) from endpoints, store them in specific data processing systems, or submit to immediate stream analysis. 
+| **[Devices provisioning and registration]({{root_url}}Programming-guide/Key-platform-features/Devices-provisioning-and-registration/)** | Enables secure authentication and registration of endpoints within a Kaa cluster.
+| **[Endpoint groups]({{root_url}}Programming-guide/Key-platform-features/Endpoint-groups/)** | Allows you to aggregate endpoints into groups and then apply other Kaa features to entire endpoint groups.
+| **[Endpoint ownership]({{root_url}}Programming-guide/Key-platform-features/Endpoint-ownership/)** | Allows you to define which users can have access to specific endpoints.
```
Comment (akokhanovskyi) [inline doc/Programming-guide/Key-platform-features/index.md]: Allows you to associate owners with specific endpoints.

File: doc/Programming-guide/Key-platform-features/index.md [hunk 3]
```diff
@@ -5,4 +5,18 @@ permalink: /:path/
...
+| **[Endpoint profiles]({{root_url}}Programming-guide/Key-platform-features/Endpoint-profiles/)** | Allows you to provide endpoints with specific attributes, which can be then used as filters for creating endpoint groups.
```
Comment (akokhanovskyi) [inline doc/Programming-guide/Key-platform-features/index.md]: "used for filters"

File: doc/Programming-guide/Key-platform-features/index.md [hunk 4]
```diff
@@ -5,4 +5,18 @@ permalink: /:path/
...
+| **[Devices provisioning and registration]({{root_url}}Programming-guide/Key-platform-features/Devices-provisioning-and-registration/)** | Enables secure authentication and registration of endpoints within a Kaa cluster.
+| **[Endpoint groups]({{root_url}}Programming-guide/Key-platform-features/Endpoint-groups/)** | Allows you to aggregate endpoints into groups and then apply other Kaa features to entire endpoint groups.
+| **[Endpoint ownership]({{root_url}}Programming-guide/Key-platform-features/Endpoint-ownership/)** | Allows you to define which users can have access to specific endpoints.
+| **[Endpoint profiles]({{root_url}}Programming-guide/Key-platform-features/Endpoint-profiles/)** | Allows you to provide endpoints with specific attributes, which can be then used as filters for creating endpoint groups. 
+| **[Events]({{root_url}}Programming-guide/Key-platform-features/Events/)** | Allows you to create custom events and make endpoints send them to some other(s) whenever a predefined condition is met.
```
Comment (akokhanovskyi) [inline doc/Programming-guide/Key-platform-features/index.md]: remove "whenever a predefined condition is met"

**Review by akokhanovskyi — CHANGES_REQUESTED** [review]

**Pull request closed without merging**